src/lib/PnP.Framework/Graph/UnifiedGroupsUtility.cs

```diff
@@ -1149,61 +996,29 @@ public static List<UnifiedGroupUser> GetUnifiedGroupMembers(UnifiedGroupEntity g
+                var users = GraphUtility.ReadPagedDataFromRequest<UnifiedGroupUser>(requestUrl, accessToken, retryCount: retryCount, delay: delay).ToList();
 
-                    // Get the members of an Office 365 group.
-                    groupUsers = await graphClient.Groups[group.GroupId].Members.Request().GetAsync();
-                    if (groupUsers.CurrentPage != null && groupUsers.CurrentPage.Count > 0)
-                    {
-                        unifiedGroupGraphUsers = new List<User>();
-
-                        GenerateGraphUserCollection(groupUsers.CurrentPage, unifiedGroupGraphUsers);
-                    }
-
-                    // Retrieve users when the results are paged.
-                    while (groupUsers.NextPageRequest != null)
-                    {
-                        groupUsers = groupUsers.NextPageRequest.GetAsync().GetAwaiter().GetResult();
-                        if (groupUsers.CurrentPage != null && groupUsers.CurrentPage.Count > 0)
-                        {
-                            GenerateGraphUserCollection(groupUsers.CurrentPage, unifiedGroupGraphUsers);
-                        }
-                    }
+                foreach (var u in users)
+                {
+                    u.UserPrincipalName ??= string.Empty;
+                    u.DisplayName ??= string.Empty;
+                    u.GivenName ??= string.Empty;
+                    u.Surname ??= string.Empty;
+                    u.Email ??= string.Empty;
+                    u.MobilePhone ??= string.Empty;
+                    u.PreferredLanguage ??= string.Empty;
+                    u.JobTitle ??= string.Empty;
+                }
 
-                    // Create the collection of type OfficeDevPnP 'UnifiedGroupUser' after all users are retrieved, including paged data.
-                    if (unifiedGroupGraphUsers != null && unifiedGroupGraphUsers.Count > 0)
-                    {
-                        unifiedGroupUsers = new List<UnifiedGroupUser>();
-                        foreach (User usr in unifiedGroupGraphUsers)
-                        {
-                            UnifiedGroupUser groupUser = new UnifiedGroupUser
-                            {
-                                Id = usr.Id,
-                                UserPrincipalName = usr.UserPrincipalName != null ? usr.UserPrincipalName : string.Empty,
-                                DisplayName = usr.DisplayName != null ? usr.DisplayName : string.Empty,
-                                GivenName = usr.GivenName != null ? usr.GivenName : string.Empty,
-                                Surname = usr.Surname != null ? usr.Surname : string.Empty,
-                                Email = usr.Mail != null ? usr.Mail : string.Empty,
-                                MobilePhone = usr.MobilePhone != null ? usr.DisplayName : string.Empty,
```

Comment:
Note the behaviour change for MobilePhone and Job Title. Is it right that these were returning the Display Name instead?